src/noit_metric.c

```diff
@@ -836,3 +842,41 @@ const char *
 noit_metric_get_full_metric_name(metric_t *m) {
   return m->expanded_metric_name ? m->expanded_metric_name : m->metric_name;
 }
+
+noit_metric_tagset_context_t *
+noit_metric_tagset_context_alloc(void) {
+  noit_metric_tagset_context_t *ctx = (noit_metric_tagset_context_t *)malloc(sizeof(*ctx));
+  ctx->refcnt = 1;
+  ctx->validate_function = NULL;
+  return ctx;
+}
+
+void
+noit_metric_tagset_context_free(noit_metric_tagset_context_t *ctx) {
+  if (ck_pr_dec_32_is_zero(&ctx->refcnt)) {
```

Comment:
why do we have a ``refcnt`` if we never increment it?

Reply:
We should have a _ref() function and we need to ref it when it is attached to the operation

Reply:
If I understand correctly: The correct usage of this would be:

alloc (set ref count to 1)
attach validation function (increments ref count to 2)
execute validation function
free (reduces coult to 1)
free (reduces count to 0, actually frees)

Reply:
added a ref function